packages/rlc-common/src/interfaces.ts

```diff
@@ -150,6 +150,11 @@ export interface RLCOptions {
   batch?: any[];
   packageDetails?: PackageDetails;
   addCredentials?: boolean;
+  /** Three possiblie values:
+   * - undefined, no credentialScopes and relevant settings would be generated
+   * - [], which means we would generate TokenCredential but no credentialScopes and relevant settings
+   * - ["..."], which means we would generate credentialScopes and relevant settings with the given values
+   */
```

Comment:
Now we handle the `undefined` and empty array as different cases.

`undefined` means we don't have OAuth2 defined;

`[]` means we have OAuth2 defined but with empty scopes.

Previously we treat the above two cases the same so all of them will not generate TokenCredential in constructor.

Reply:
This is another place where we should consider if we are generating an Azure SDK or not since the `{endpoint}/.default` value only makes sense for Azure services.

Reply:
Haha, there may be another question - how to handle the empty scopes for non-Azure? How about explicitly set this as an empty array?

```
credentialScopes: []
```